### ISSUE TITLE
Handsdown crashes with a `UnicodeDecodeError` on windows due to the default encoding not being utf8

### ISSUE THREAD

**Issue by FredHappyface** [vemel/handsdown#16]
Hi thanks for this library

**Describe the bug**
Handsdown crashes with a `UnicodeDecodeError` on windows due to the default encoding not being utf8

**To Reproduce**
1. Create a readme file containing utf-8 on windows (tested with https://raw.githubusercontent.com/FHPythonUtils/AnsiToImg/8d52d50caa0787ceb33d2db2f81880c565262790/README.md)
2. Run handsdown --cleanup
3. Result:
    ```
    handsdown --cleanup
    Traceback (most recent call last):
      File "C:\Python310\lib\runpy.py", line 196, in _run_module_as_main
        return _run_code(code, main_globals, None,
      File "C:\Python310\lib\runpy.py", line 86, in _run_code
        exec(code, run_globals)
      File "C:\Python310\lib\site-packages\handsdown\main.py", line 53, in main
        generator = Generator(
      File "C:\Python310\lib\site-packages\handsdown\generator.py", line 101, in __init__
        self._prepare_index()
      File "C:\Python310\lib\site-packages\handsdown\generator.py", line 110, in _prepare_index
        self.md_index.read(readme_path)
      File "C:\Python310\lib\site-packages\handsdown\md_document.py", line 94, in read
        self._content = path.read_text()
      File "C:\Python310\lib\pathlib.py", line 1133, in read_text
        return f.read()
      File "C:\Python310\lib\encodings\cp1252.py", line 23, in decode
        return codecs.charmap_decode(input,self.errors,decoding_table)[0]
    UnicodeDecodeError: 'charmap' codec can't decode byte 0x8f in position 3826: character maps to <undefined>
    ```

**Expected behavior**
Documentation to be generated without error 

**Desktop (please complete the following information):**
 - OS: Windows 10 21H1
 - Version 0.5.1

**Additional context**
This is due to omitting an encoding for read_text and write_text

Happy to make a pr for this. Cheers for your time :)

**Comment by vemel** [comment]
Hello! Thanks for the report.

I am not using Windows for development, so I would highly appreciate if you create a PR. I wonder how many places in my code are affected.

I am going to test it tomorrow on Windows.

**Comment by vemel** [comment]
The fix has been included in `handsdown 0.6.0` release. Please let me know if it works as it should.

**Comment by FredHappyface** [comment]
Hi, thanks so much for the fix! Works well, just have a few questions which I'll open new issues for

Thank you :)
